python/ray/train/tests/test_gpu.py

```diff
@@ -100,6 +100,52 @@ def train_fn():
         )
 
 
+@pytest.mark.parametrize("cuda_visible_devices", ["", "1,2"])
+@pytest.mark.parametrize("num_gpus_per_worker", [0.5, 1, 2])
+def test_torch_get_amd_device(
+    shutdown_only, num_gpus_per_worker, cuda_visible_devices, monkeypatch, tmp_path
+):
+    if cuda_visible_devices:
+        # Test if `get_device` is correct even with user specified env var.
+        monkeypatch.setenv("ROCR_VISIBLE_DEVICES", cuda_visible_devices)
```

Comment:
Can you clear CUDA_VISIBLE_DEVICES to ensure that we are indeed reading from ROCR_VISIBLE_DEVICES? 

Also it'd be better to check if we are actually using an `AMDGPUAcceleratorManager`, or fallback to `NvidiaGPUAcceleratorManager` [here](https://github.com/ray-project/ray/blob/cbdbd7537db8a4b774c3cfcc2f72a346bbd97a24/python/ray/_private/accelerators/__init__.py#L54).